### PR TITLE
Added Array Left Rotation

### PR DESCRIPTION
**Pull request by sangin-lee**
### **Describe your change:**
Added Array Left Rotation.
A left rotation operation on an array shifts each of the array's elements given integer n unit to the left.

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### **Checklist:**

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new Java files are placed inside an existing directory.
- [ ] All filenames are in all uppercase characters with no spaces or dashes.
- [x] All functions and variable names follow Java naming conventions.
- [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: src/main/java/com/thealgorithms/others/ArrayLeftRotation.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	public static void printIntegerArray(int[] arr) {
+		int size = arr.length;
+		String str = "[ ";
+		for (int i = 0; i < size; i++) {
+			str += arr[i] + " ";
+		}
+		str += "]";
+		System.out.println(str);
+	}
+
```
Comment (siriak) [inline src/main/java/com/thealgorithms/others/ArrayLeftRotation.java:44]: Printing is not necessary

File: src/main/java/com/thealgorithms/others/ArrayLeftRotation.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	public static void main(String[] args) {
```
Comment (siriak) [inline src/main/java/com/thealgorithms/others/ArrayLeftRotation.java:45]: Please also remove main() and write a couple of tests instead as in https://github.com/TheAlgorithms/Java/blob/66d8d51de67fd3e7c7249930f4a912ad9560d823/src/test/java/com/thealgorithms/searches/QuickSelectTest.java

**Review by siriak — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/others/ArrayLeftRotation.java (hunk 2 above)
Comment (sangin-lee) [reply src/main/java/com/thealgorithms/others/ArrayLeftRotation.java]: @siriak I wrote some junit tests, but I have no idea of why build fails.

> package org.junit.jupiter.api does not exist

I wrote the test just like QuickSelectTest.java you've commented.
Can you give me some advice?

Comment (sangin-lee) [reply src/main/java/com/thealgorithms/others/ArrayLeftRotation.java]: I resolved it. Thanks 👍

**Review by siriak — APPROVED** [review]
Awesome, thanks!

**Comment by sangin-lee** [comment]
Thanks for merging

**Pull request merged**